idx/elasticsearch/elasticsearch.go

```diff
@@ -0,0 +1,441 @@
+		                "dynamic_templates": [
+		                    {
+		                        "strings": {
+		                            "mapping": {
+		                                "index": "not_analyzed",
+		                                "type": "string"
+		                            },
+		                            "match_mapping_type": "string"
+		                        }
+		                    }
+		                ],
+		                "_all": {
+		                    "enabled": false
+		                },
+		                "_timestamp": {
+		                    "enabled": false
+		                },
+		                "properties": {
+		                    "id": {
+		                        "type": "string",
+		                        "index": "not_analyzed"
+		                    },
+		                    "interval": {
+		                        "type": "long"
+		                    },
+		                    "lastUpdate": {
+		                        "type": "long"
+		                    },
+		                    "metric": {
+		                        "type": "string",
+		                        "index": "not_analyzed"
+		                    },
+		                    "name": {
+		                        "type": "string",
+		                        "index": "not_analyzed"
+		                    },
+		                    "node_count": {
+		                        "type": "long"
+		                    },
+		                    "org_id": {
+		                        "type": "long"
+		                    },
+		                    "tags": {
+		                        "type": "string",
+		                        "index": "not_analyzed"
+		                    },
+		                    "mtype": {
+		                        "type": "string",
+		                        "index": "not_analyzed"
+		                    },
+		                    "unit": {
+		                        "type": "string",
+		                        "index": "not_analyzed"
+		                    }
+		                }
+					}
+				}
+			}`
+
+			_, err = e.Conn.DoCommand("PUT", fmt.Sprintf("/%s", Index), nil, metricMapping)
+			if err != nil {
+				return err
+			}
+			// if we dont wait here, rebuildIndex will fail due to ES returing a 503.
+			time.Sleep(time.Second)
+		}
+	}
+	log.Info("Setting up ES bulkIndexer")
+	e.BulkIndexer = e.Conn.NewBulkIndexer(MaxConns)
+
+	//dont retry sends.
+	e.BulkIndexer.RetryForSeconds = 0
+
+	// index at most MaxBufferDocs per request.
+	e.BulkIndexer.BulkMaxDocs = MaxBufferDocs
+	//Setting BulkMaxBuffer to 1KB per doc ensure we will hold at least MaxBufferDocs before flushing
+	e.BulkIndexer.BulkMaxBuffer = 1024 * MaxBufferDocs
+
+	//flush at least every 10seconds.
+	bufferDelay, err := time.ParseDuration(BufferDelayMax)
+	if err != nil {
+		return err
+	}
+	e.BulkIndexer.BufferDelayMax = bufferDelay
+	e.BulkIndexer.Refresh = true
+	e.BulkIndexer.Sender = e.getBulkSend()
+
+	interval, err := time.ParseDuration(RetryInterval)
```

Comment:
you can skip this step by declaring the flag as https://golang.org/pkg/flag/#DurationVar